<a id="top"></a>
# [Technical Derivations: Projective Process Monism](#top)
**Every numerical claim in the paper, reproduced from first principles.**

Each section references the corresponding paper section. All computations use the `ppm/` package — no hardcoded numerics except observed values (PDG/Planck).

---

## Table of Contents

| § | Topic | Key result |
|---|-------|------------|
| 1 | [CP³ spectral data](#s1-spectral) | $\lambda_k = k(k+3)$, multiplicities $d_k$, $\tau$-traces |
| 2 | [Pyramidal numbers](#s2-pyramidal) | $P_3^2\ln\varphi \approx P_4\pi$ (0.074%) |
| 3 | [$\alpha$ Route I: spectral](#s3-alpha-route-i) | $1/\alpha = 137.257$ (0.16%) |
| 4 | [$\alpha$ Route II: cogito](#s4-alpha-route-ii) | $1/\alpha \approx 137.6$ (0.4%) |
| 5 | [$\alpha$ Route III: instanton](#s5-alpha-route-iii) | $S = 30\pi$, prefactor open |
| 6 | [CP$^n$ selectivity](#s6-cpn-selectivity) | Only $n=3$ gives $\sim137$ |
| 7 | [Energy hierarchy](#s7-hierarchy) | $g = 2\pi$, all particle masses |
| 8 | [Gauge structure](#s8-gauge) | $\sin^2\theta_W = 3/8$, 3 generations |
| 9 | [Higgs sector](#s9-higgs) | $\lambda = 1/(4\sqrt\pi)$, $y_t$, $\Delta\lambda$ |
| 10 | [Lepton masses](#s10-leptons) | $(2\pi)^{3/2}$, $(3/2)e^{\pi^2/2}$ |
| 11 | [CKM & CP violation](#s11-ckm) | $\delta_{CP} = \pi/\varphi^2 \approx 68.5°$ |
| 12 | [PMNS & neutrinos](#s12-pmns) | TBM, $\theta_{\rm strong} = 0$, sterile $\nu$ |
| 13 | [Heat kernel & spectral](#s13-heat-kernel) | $\zeta_\Delta(0) = -733/945$, $Z_1$ |
| 14 | [Instanton sector](#s14-instanton) | 30 zero modes, $\varphi^{-196}$, $Z_{T^2}$ |
| 15 | [Golden ratio](#s15-golden-ratio) | $A_5$ decomposition, L-function |
| 16 | [Cosmology](#s16-cosmology) | $\Lambda$, $H_0$, $G_{\rm eff}(z)$ |
| 16b | [Consciousness numerics](#s16b-consciousness) | $\Phi$, decoherence times, k-window |
| F | [Falsification tests](#falsification-tests) | structural counterfactuals |

In [1]:
import sys, os, math
import numpy as np
import matplotlib
matplotlib.use('Agg')
import matplotlib.pyplot as plt
import pandas as pd

# Ensure ppm/ is importable from notebooks/ directory
sys.path.insert(0, os.path.abspath('..'))

import ppm
from ppm import constants as C
from ppm import hierarchy as H
from ppm import alpha as A
from ppm import gauge as G
from ppm import higgs as HI
from ppm import instanton as I
from ppm import spectral as S
from ppm import cosmology as GR
from ppm import golden_ratio as GR_phi
from ppm import berry_phase as BP
from ppm import neutrino as NU
from ppm import predictions as PRED
from ppm import verify as V

import warnings
warnings.filterwarnings('ignore')

print(f"ppm v{ppm.__version__} loaded — {len(dir(ppm))} submodules")


ppm v2.0.0 loaded — 35 submodules


<a id="s1-spectral"></a>
## [§ 1 — CP³ Spectral Data](#s1-spectral)

**Paper §5, Appendix A.**

Scalar Laplacian on $\mathbb{CP}^3$ (Fubini–Study metric, $c=1$):

$$\lambda_k = k(k+3), \quad k = 0, 1, 2, \ldots$$

Total multiplicity: $d_k = \binom{k+3}{3}^2 - \binom{k+2}{3}^2$

$\tau$-trace (even under anti-holomorphic involution): $\mathrm{tr}(\tau|V_k) = \binom{k+3}{3} - \binom{k+2}{3}$

In [2]:
print("CP³ Laplacian Spectrum (first 15 levels)")
print("=" * 72)
print(f"{'k':>3}  {'λ_k':>6}  {'d_k':>8}  {'tr(τ|V_k)':>10}  {'d_k / tr(τ)':>12}  {'Σd_k':>10}")
print("-" * 72)

cum_d = 0
for k in range(15):
    lam_k = k * (k + 3)
    d_k = math.comb(k+3, 3)**2 - math.comb(k+2, 3)**2
    tr_tau = math.comb(k+3, 3) - math.comb(k+2, 3)
    cum_d += d_k
    ratio = d_k / tr_tau if tr_tau > 0 else float('inf')
    print(f"{k:3d}  {lam_k:6d}  {d_k:8d}  {tr_tau:10d}  {ratio:12.1f}  {cum_d:10d}")

print()
print("Note: d_k grows as k^5, tr(τ) as k^2 → ratio Θ_{CP³}/Θ^τ → large → α small")


CP³ Laplacian Spectrum (first 15 levels)
  k     λ_k       d_k   tr(τ|V_k)   d_k / tr(τ)        Σd_k
------------------------------------------------------------------------
  0       0         1           1           1.0           1
  1       4        15           3           5.0          16
  2      10        84           6          14.0         100
  3      18       300          10          30.0         400
  4      28       825          15          55.0        1225
  5      40      1911          21          91.0        3136
  6      54      3920          28         140.0        7056
  7      70      7344          36         204.0       14400
  8      88     12825          45         285.0       27225
  9     108     21175          55         385.0       48400
 10     130     33396          66         506.0       81796
 11     154     50700          78         650.0      132496
 12     180     74529          91         819.0      207025
 13     208    106575         105        1015.

<a id="s2-pyramidal"></a>
## [§ 2 — Pyramidal Number Structure](#s2-pyramidal)

**Paper §8 (Exact Predictions), section-gravity.tex (Why $\varphi$).**

Square pyramidal numbers: $P_n = n(n+1)(2n+1)/6$

The central near-identity: $P_3^2 \cdot \ln\varphi \approx P_4 \cdot \pi$ (0.074% mismatch).

Equivalently: $e^{-30\pi} \approx \varphi^{-196}$.

Both 196 and 30 are pyramidal: $P_3^2 = 14^2 = 196$, $P_4 = 30$.

In [3]:
print("Square Pyramidal Numbers P_n = n(n+1)(2n+1)/6")
print("=" * 50)
for n in range(1, 8):
    print(f"  P_{n} = {GR_phi.pyramidal_number(n)}")

pi = GR_phi.pyramidal_identity()
print()
print(f"P₃ = {pi['P3']},  P₃² = {pi['P3_squared']}")
print(f"P₄ = {pi['P4']}")
print()
print(f"P₃² · ln(φ) = {pi['P3_squared']} × {math.log(C.PHI):.5f} = {pi['lhs']:.4f}")
print(f"P₄  · π     = {pi['P4']} × {math.pi:.5f} = {pi['rhs']:.4f}")
print(f"Ratio = {pi['ratio']:.6f}  (mismatch = {pi['mismatch_pct']:.4f}%)")


Square Pyramidal Numbers P_n = n(n+1)(2n+1)/6
  P_1 = 1
  P_2 = 5
  P_3 = 14
  P_4 = 30
  P_5 = 55
  P_6 = 91
  P_7 = 140

P₃ = 14,  P₃² = 196
P₄ = 30

P₃² · ln(φ) = 196 × 0.48121 = 94.3175
P₄  · π     = 30 × 3.14159 = 94.2478
Ratio = 1.000740  (mismatch = 0.0740%)


<a id="s3-alpha-route-i"></a>
## [§ 3 — $\alpha$ Route I: Twisted Heat Trace](#s3-alpha-route-i)

**Paper §5 (Three Derivations of $\alpha$).**

$$\alpha = \frac{\Theta^\tau(t^*)}{\Theta_{\mathbb{CP}^3}(t^*)} \bigg|_{t^* = 1/32}$$

Half-variance condition: $t^* = 1/(2(n+1)^2)$ for $\mathbb{CP}^n$. For $n=3$: $t^* = 1/32$.

**Result:** $1/\alpha = 137.257$ (error $-0.161\%$ from 137.036).

In [4]:
ts = A.t_star(n=3)
print(f"t* = 1/(2×(3+1)²) = 1/32 = {ts:.6f}")
print()

# Show convergence
print("Convergence of Route I with nmax:")
print(f"{'nmax':>6}  {'1/α':>12}  {'error %':>10}")
print("-" * 32)
for nmax in [10, 20, 50, 100, 200]:
    r = A.alpha_from_spectral_geometry(nmax)
    print(f"{nmax:6d}  {r['alpha_inv']:12.4f}  {r['error_pct']:+10.4f}%")

print()
print("Converges by nmax ≈ 50. Using nmax=200:")
r1 = A.alpha_from_spectral_geometry(200)
print(f"  1/α = {r1['alpha_inv']:.4f}")
print(f"  error = {r1['error_pct']:+.4f}%")
print(f"  status: {r1['status']}")


t* = 1/(2×(3+1)²) = 1/32 = 0.031250

Convergence of Route I with nmax:
  nmax           1/α     error %
--------------------------------
    10      105.8379    +29.4773%
    20      137.2430     -0.1508%
    50      137.2573     -0.1612%
   100      137.2573     -0.1612%
   200      137.2573     -0.1612%

Converges by nmax ≈ 50. Using nmax=200:
  1/α = 137.2573
  error = -0.1612%
  status: COMPLETE


In [5]:
# Show the actual heat trace values at t*
theta_tau, theta_cp3 = A._twisted_heat_traces(ts, nmax=200)
print(f"Θ^τ(t*)    = {theta_tau:.6f}")
print(f"Θ_CP³(t*)  = {theta_cp3:.6f}")
print(f"ratio      = {theta_tau/theta_cp3:.8f}")
print(f"1/ratio    = {theta_cp3/theta_tau:.4f}")
print()

# Partial sums to show convergence
print("Partial sums (first 10 terms):")
print(f"{'k':>3}  {'λ_k':>6}  {'w=e^(-λt)':>14}  {'tr·w':>14}  {'d·w':>14}  {'running 1/α':>12}")
print("-" * 70)
running_tau = 0.0
running_cp3 = 0.0
for k in range(10):
    lam_k = k * (k + 3)
    tr_tau = math.comb(k+3, 3) - math.comb(k+2, 3)
    d_k = math.comb(k+3, 3)**2 - math.comb(k+2, 3)**2
    w = math.exp(-lam_k * ts)
    running_tau += tr_tau * w
    running_cp3 += d_k * w
    ratio = running_cp3 / running_tau if running_tau > 0 else float('inf')
    print(f"{k:3d}  {lam_k:6d}  {w:14.6e}  {tr_tau*w:14.6e}  {d_k*w:14.6e}  {ratio:12.3f}")


Θ^τ(t*)    = 42.355208
Θ_CP³(t*)  = 5813.561379
ratio      = 0.00728559
1/ratio    = 137.2573

Partial sums (first 10 terms):
  k     λ_k       w=e^(-λt)            tr·w             d·w   running 1/α
----------------------------------------------------------------------
  0       0    1.000000e+00    1.000000e+00    1.000000e+00         1.000
  1       4    8.824969e-01    2.647491e+00    1.323745e+01         3.903
  2      10    7.316156e-01    4.389694e+00    6.145571e+01         9.418
  3      18    5.697828e-01    5.697828e+00    1.709348e+02        17.956
  4      28    4.168620e-01    6.252930e+00    3.439112e+02        29.545
  5      40    2.865048e-01    6.016601e+00    5.475107e+02        43.763
  6      54    1.849814e-01    5.179479e+00    7.251271e+02        59.748
  7      70    1.121969e-01    4.039088e+00    8.239740e+02        76.289
  8      88    6.392786e-02    2.876754e+00    8.198748e+02        92.048
  9     108    3.421812e-02    1.881997e+00    7.245687e+02    

<a id="s4-alpha-route-ii"></a>
## [§ 4 — $\alpha$ Route II: Consistency Check (G-Inversion)](#s4-alpha-route-ii)

**Paper §5 (consistency check, not an independent derivation).**

Inverting the gravitational self-coupling formula for $G$ recovers $\alpha$ from $G_{\rm obs}$, $m_\pi$, and $N_\infty = \varphi^{392}$:

$$\alpha = \frac{G_{\rm obs}\, m_\pi^2 \sqrt{N_\infty}}{16\pi^4 \hbar c}$$

This is *not* a third independent derivation of $\alpha$. It is a self-consistency check: the $G$ formula derived from the framework must invert to a value of $\alpha$ that agrees with Route I's spectral-geometry result. It does, to $0.4\%$. The check uses the observed $G$ as input, so it cannot be falsified by observations of $\alpha$ alone — but a divergence between the two routes would indicate the framework's $G$ and $\alpha$ derivations are inconsistent.

**Result:** $1/\alpha \approx 137.6$ ($0.4\%$ from Route I's $137.257$).

In [6]:
r2 = A.alpha_from_cogito_loop()
print("Route II: Cogito Loop")
print("=" * 60)
print(f"  N       = {r2['N']:.4e}")
print(f"  √N      = {r2['sqrt_N']:.4e}")
print(f"  φ^196   = {r2['phi_196']:.4e}")
print(f"  √N/φ¹⁹⁶ = {r2['sqrt_N_over_phi196']:.6f}")
print()
print(f"  1/α     = {r2['alpha_inv']:.4f}")
print(f"  error   = {r2['error_pct']:+.3f}%")
print(f"  status  = {r2['status']}")


Route II: Cogito Loop
  N       = 8.5931e+81
  √N      = 9.2699e+40
  φ^196   = 9.1533e+40
  √N/φ¹⁹⁶ = 1.012736

  1/α     = 137.5564
  error   = -0.378%
  status  = PARTIAL (anchored to G_obs, Λ_obs)


<a id="s5-alpha-route-iii"></a>
## [§ 5 — $\alpha$ Route III: Instanton](#s5-alpha-route-iii)

**Paper §8, Appendix A.**

$$S = (N{-}1) \cdot r^2 \cdot \pi = 3 \times 10 \times \pi = 30\pi$$

Bare rate: $R_\tau^{\rm bare} = \tfrac{1}{2} e^{-30\pi} \approx 5.86 \times 10^{-42}$.

Prefactor from 30 zero modes needed to bridge to $\alpha \approx 7.3 \times 10^{-3}$.

**Status: PARKED** — prefactor calculation requires fluctuation spectrum (FFS blocker).

In [7]:
r3 = A.alpha_from_instanton()
print("Route III: Instanton")
print("=" * 60)
print(f"  S_inst         = {r3['S_inst']:.4f} = 30π")
print(f"  R_τ(bare)      = {r3['R_tau_bare']:.4e}")
print(f"  α(observed)    = {r3['alpha_observed']:.6e}")
print(f"  prefactor gap  = {r3['prefactor_gap']:.3e}  (~10^{r3['log10_prefactor_gap']:.1f})")
print(f"  status         = {r3['status']}")
print()

# Compare all three routes
print("Summary of Three Routes")
print("-" * 60)
print(f"{'Route':>10}  {'1/α':>10}  {'error':>10}  {'status':>20}")
print("-" * 60)
r1 = A.alpha_from_spectral_geometry(200)
r2 = A.alpha_from_cogito_loop()
print(f"{'I':>10}  {r1['alpha_inv']:10.3f}  {r1['error_pct']:+9.3f}%  {'COMPLETE':>20}")
print(f"{'II':>10}  {r2['alpha_inv']:10.3f}  {r2['error_pct']:+9.3f}%  {'PARTIAL':>20}")
print(f"{'III':>10}  {'—':>10}  {'—':>10}  {'PARKED':>20}")
print(f"{'Observed':>10}  {C.ALPHA_EM_INV:10.3f}")


Route III: Instanton
  S_inst         = 94.2478 = 30π
  R_τ(bare)      = 5.8571e-42
  α(observed)    = 7.297353e-03
  prefactor gap  = 1.246e+39  (~10^39.1)
  status         = PARKED (prefactor open)

Summary of Three Routes
------------------------------------------------------------
     Route         1/α       error                status
------------------------------------------------------------
         I     137.257     -0.161%              COMPLETE
        II     137.556     -0.378%               PARTIAL
       III           —           —                PARKED
  Observed     137.036


<a id="s6-cpn-selectivity"></a>
## [§ 6 — CP$^n$ Selectivity](#s6-cpn-selectivity)

**Paper §5.**

The half-variance condition $t^* = 1/(2(n{+}1)^2)$ applied to $\mathbb{CP}^n$ for $n = 1, \ldots, 7$:

Only $n=3$ gives $1/\alpha$ in the physical range ($\sim 137$).

In [8]:
fam = A.alpha_cpn_family(range(1, 8), nmax=300)

print("CP^n Selectivity: 1/α from half-variance condition")
print("=" * 50)
print(f"{'n':>3}  {'t*':>12}  {'1/α':>12}  {'physical?':>10}")
print("-" * 42)
for n in range(1, 8):
    ts = fam[n]['t_star']
    ai = fam[n]['alpha_inv']
    phys = "✓ YES" if 130 < ai < 145 else ""
    print(f"{n:3d}  {ts:12.6f}  {ai:12.1f}  {phys:>10}")

# Plot
fig, ax = plt.subplots(figsize=(10, 5))
ns = list(range(1, 8))
alphas = [fam[n]['alpha_inv'] for n in ns]
ax.semilogy(ns, alphas, 'bo-', markersize=10, linewidth=2, label=r'$1/\alpha$ from CP$^n$')
ax.axhline(137.036, color='red', linestyle='--', linewidth=1.5, label='Observed 1/α = 137.036')
ax.axhspan(130, 145, alpha=0.1, color='red')
ax.set_xlabel('n (dimension of CP$^n$)')
ax.set_ylabel(r'$1/\alpha$')
ax.set_title(r'CP$^n$ Selectivity: Only $n=3$ Gives Physical $\alpha$')
ax.set_xticks(ns)
ax.legend()
ax.grid(True, alpha=0.3)
plt.tight_layout()
plt.show()


CP^n Selectivity: 1/α from half-variance condition
  n            t*           1/α   physical?
------------------------------------------
  1      0.125000           3.2            
  2      0.055556          18.2            
  3      0.031250         137.3       ✓ YES
  4      0.020000        1258.5            
  5      0.013889       13314.0            
  6      0.010204      157680.5            
  7      0.007812     2049280.9            


<a id="s7-hierarchy"></a>
## [§ 7 — Energy Hierarchy](#s7-hierarchy)

**Paper §4 (Bootstrap & Hierarchy Scaling).**

$$E(k) = 140\,\text{MeV} \times (2\pi)^{(51-k)/2}$$

Topological coupling $g = 2\pi$ from two independent proofs:
1. $g^2 = |Z_2 \times Z_2| \times \text{Vol}(\text{RP}^3) = 4\pi^2 \;\Rightarrow\; g = 2\pi$
2. Maslov area: $A_{\min} = (N{+}1)\pi/2 = 2\pi$ for $N=3$

In [9]:
gt = H.g_from_topology()
print("g = 2π from topology")
print("=" * 60)
print(f"  g (topological product) = {gt['g_topo']:.6f}")
print(f"  g (Maslov area)         = {gt['g_maslov']:.6f}")
print(f"  2π                      = {2*math.pi:.6f}")
print(f"  g (empirical best-fit)  = {gt['g_empirical']}")
print(f"  error vs empirical      = {gt['error_pct']:+.2f}%")


g = 2π from topology
  g (topological product) = 6.283185
  g (Maslov area)         = 6.283185
  2π                      = 6.283185
  g (empirical best-fit)  = 6.32
  error vs empirical      = +0.59%


In [10]:
print("Full Particle Mass Hierarchy (bare E(k))")
print("=" * 85)
print("Note: bare E(k) is the geometric position on the cascade ladder.")
print("Physical particle masses use the EWSB-anchored forms (top via π·E(k_EWSB),")
print("Higgs via v√(2λ), light leptons via Bohr-Sommerfeld); see predictions.ipynb")
print("for the comparable-to-observed table.")
print("=" * 85)
print(f"{'Particle':>12}  {'k':>6}  {'E(k) GeV':>14}  {'Observed GeV':>14}  {'Cat':>10}")
print("-" * 85)

table = H.k_level_table()
for row in table:
    obs = f"{row['mass_observed_GeV']:.4e}" if row['mass_observed_GeV'] else "—"
    print(f"{row['name']:>12}  {row['k']:6.2f}  {row['E_predicted_GeV']:14.4e}  {obs:>14}  {row['category']:>10}")

# Planck anchor — bare hierarchy at k=1
pa = H.planck_anchor()
print(f"\nPlanck anchor: E(k=1) = {pa['E_predicted_GeV']:.3e} GeV  (obs: {pa['E_observed_GeV']:.2e}, err: {pa['error_pct']:+.1f}%)")
print("(Bare-hierarchy match at k=1 is what fixes the cascade base.)")


Full Particle Mass Hierarchy (bare E(k))
Note: bare E(k) is the geometric position on the cascade ladder.
Physical particle masses use the EWSB-anchored forms (top via π·E(k_EWSB),
Higgs via v√(2λ), light leptons via Bohr-Sommerfeld); see predictions.ipynb
for the comparable-to-observed table.
    Particle       k        E(k) GeV    Observed GeV         Cat
-------------------------------------------------------------------------------------
      Planck    1.00      1.2607e+19      1.2200e+19       scale
 UV boundary   10.00      3.2271e+15               —       scale
  Pati-Salam   16.25      1.0340e+13               —       scale
         top   44.50      5.4981e+01      1.7270e+02       quark
       Higgs   44.50      5.4981e+01      1.2525e+02       boson
           W   44.50      5.4981e+01      8.0377e+01       boson
           Z   44.50      5.4981e+01      9.1188e+01       boson
      bottom   46.00      1.3854e+01      4.1800e+00       quark
       charm   47.50      3.4909e+

<a id="s8-gauge"></a>
## [§ 8 — Gauge Structure](#s8-gauge)

**Paper §6 (Gauge Structure).**

Pati–Salam embedding: $\text{SU}(4) \to \text{SU}(3) \times \text{SU}(2) \times \text{U}(1)$

$$\alpha_{\rm GUT} = 1/r^2 = 1/10, \qquad \sin^2\theta_W\big|_{k_{\rm break}} = 3/8$$

SM one-loop running from $M_Z$ to $E_{\rm break}$: $\sin^2\theta_W(E_{\rm break}) = 0.3755$ (0.13% from 3/8).

Generations: $N_{\rm gen} = 3$ from $\mathbb{CP}^3$ topology.

In [11]:
# sin²θ_W
stw = G.sin2_theta_W_sm_running()
print("Weinberg Angle")
print("=" * 60)
print(f"  PPM (Pati-Salam):  sin²θ_W = {stw['sin2_tW_ppm']:.5f} = 3/8")
print(f"  SM running to E_break:       {stw['sin2_tW_sm']:.5f}")
print(f"  Agreement:                    {stw['agreement_pct']:.3f}%")
print(f"  E_break =                     {stw['E_break_GeV']:.3e} GeV")
print()

# Coupling comparison at E_break
coup = G.couplings_at_ebreak()
print("Couplings at E_break (PPM vs SM running):")
print(f"  α₁: PPM={coup['alpha1_ppm']:.5f}  SM={coup['alpha1_sm']:.5f}  ratio={coup['ratio1']:.2f}")
print(f"  α₂: PPM={coup['alpha2_ppm']:.5f}  SM={coup['alpha2_sm']:.5f}  ratio={coup['ratio2']:.2f}")
print(f"  α₃: PPM={coup['alpha3_ppm']:.5f}  SM={coup['alpha3_sm']:.5f}  ratio={coup['ratio3']:.2f}")
print(f"  Note: {coup['note'][:80]}")
print()

# Generations
gen = G.generation_count()
print(f"Generations: N_gen = {gen['N_generations']}  ({gen['note']})")


Weinberg Angle
  PPM (Pati-Salam):  sin²θ_W = 0.37500 = 3/8
  SM running to E_break:       0.37549
  Agreement:                    0.130%
  E_break =                     1.034e+13 GeV

Couplings at E_break (PPM vs SM running):
  α₁: PPM=0.16667  SM=0.02361  ratio=7.06
  α₂: PPM=0.10000  SM=0.02356  ratio=4.24
  α₃: PPM=0.10000  SM=0.02709  ratio=3.69
  Note: Factor ~4× normalization mismatch; root cause: Fubini-Study vs MSbar; resolution

Generations: N_gen = 3  (From CP³ topology; exact cohomology argument)


<a id="s9-higgs"></a>
## [§ 9 — Higgs Sector](#s9-higgs)

**Paper §2 ($\tau$ involution), §7 (SM parameters).**

$$\lambda_{\rm PPM} = \frac{1}{4\sqrt\pi} \approx 0.14105$$

Geometric identity: $\Delta\lambda = \lambda_{\rm PPM} - \lambda_\tau = \frac{1}{2\sqrt\pi}$

SM running brackets this: 1-loop gives 0.270 (95.8%), 2-loop gives 0.286 (101.5%).

$$y_t^{\rm PPM} = \frac{\pi}{2(2\pi)^{1/4}} \approx 0.992$$

In [12]:
lam = HI.lambda_ppm()
lam_tau = HI.lambda_tau_conjugate()
dl = HI.delta_lambda()
yt = HI.top_yukawa_ppm()
comp = HI.higgs_quartic_comparison()
gi = HI.geometric_identity_check()
bl = HI.beta_lambda_ppm()

print("Higgs Sector Derivations")
print("=" * 60)
print(f"  λ_PPM = 1/(4√π) = {lam:.6f}")
print(f"  λ_τ   = -λ_PPM  = {lam_tau:.6f}")
print(f"  Δλ    = 1/(2√π) = {dl:.6f}")
print()
print(f"  SM comparison:")
print(f"    1-loop Δλ_SM = {gi['delta_lambda_observed_1loop']:.3f}  ({gi['sm_match_pct_1loop']:.1f}% from geometric)")
print(f"    2-loop Δλ_SM = {gi['delta_lambda_observed_2loop']:.3f}  ({gi['sm_match_pct_2loop']:.1f}% from geometric)")
print()
print(f"  λ_PPM vs observed: {comp['lambda_ppm']:.5f} vs {comp['lambda_observed_MZ']:.4f} ({comp['error_pct']:+.1f}%)")
print()

v = 246.22
m_H = v * math.sqrt(2 * lam)
print(f"  m_H = v√(2λ) = {m_H:.1f} GeV  (observed: 125.25 GeV, {abs(m_H/125.25-1)*100:.1f}%)")
print()
print(f"  y_t = π/(2(2π)^{{1/4}}) = {yt:.6f}  (observed: {C.Y_TOP_OBSERVED})")
m_t = yt * v / math.sqrt(2)
print(f"  m_t = y_t × v/√2 = {m_t:.1f} GeV  (observed: {C.M_TOP_GEV})")
print()
print(f"  β_λ at PPM point = {bl['beta_lambda']:.5f}")


Higgs Sector Derivations
  λ_PPM = 1/(4√π) = 0.141047
  λ_τ   = -λ_PPM  = -0.141047
  Δλ    = 1/(2√π) = 0.282095

  SM comparison:
    1-loop Δλ_SM = 0.270  (4.3% from geometric)
    2-loop Δλ_SM = 0.286  (1.4% from geometric)

  λ_PPM vs observed: 0.14105 vs 0.1292 (+9.2%)

  m_H = v√(2λ) = 130.8 GeV  (observed: 125.25 GeV, 4.4%)

  y_t = π/(2(2π)^{1/4}) = 0.992144  (observed: 0.992)
  m_t = y_t × v/√2 = 172.7 GeV  (observed: 172.7)

  β_λ at PPM point = -0.02542


<a id="s10-leptons"></a>
## [§ 10 — Lepton Mass Ratios](#s10-leptons)

**Paper §7 (SM Parameters).**

$$m_\tau / m_\mu \approx (2\pi)^{3/2} = 15.75 \quad (\text{observed: } 16.82, \text{ error } {-}6.3\%)$$

$$m_\mu / m_e \approx \tfrac{3}{2} e^{\pi^2/2} = 208.6 \quad (\text{observed: } 206.8, \text{ error } {+}0.9\%)$$

In [13]:
lmr = G.lepton_mass_ratios()
print("Lepton Mass Ratios")
print("=" * 60)
print(f"  m_τ/m_μ:")
print(f"    PPM: (2π)^{{3/2}} = {lmr['tau_mu_ppm']:.3f}")
print(f"    Obs:              {lmr['tau_mu_obs']:.3f}")
print(f"    Error:            {lmr['tau_mu_err_pct']:+.1f}%")
print()
print(f"  m_μ/m_e:")
print(f"    PPM: (3/2)e^{{π²/2}} = {lmr['mu_e_ppm']:.2f}")
print(f"    Obs:                {lmr['mu_e_obs']:.2f}")
print(f"    Error:              {lmr['mu_e_err_pct']:+.1f}%")
print()

# α from blanket volume (independent route)
abv = G.alpha_blanket_volume()
print("  Blanket volume route to α:")
print(f"    1/α (blanket) = e^{{π²/2}} = {abv['alpha_inv_blanket']:.1f}  (error {abv['err_blanket_pct']:+.1f}%)")
print(f"    1/α (heat kernel) = {abv['alpha_inv_heat_kernel']:.1f}  (error {abv['err_hk_pct']:+.1f}%)")
print(f"    Weighted average = {abv['weighted_average']:.1f}  (error {abv['err_weighted_pct']:+.1f}%)")


Lepton Mass Ratios
  m_τ/m_μ:
    PPM: (2π)^{3/2} = 15.750
    Obs:              16.817
    Error:            -6.3%

  m_μ/m_e:
    PPM: (3/2)e^{π²/2} = 208.57
    Obs:                206.77
    Error:              +0.9%

  Blanket volume route to α:
    1/α (blanket) = e^{π²/2} = 139.0  (error +1.5%)
    1/α (heat kernel) = 136.8  (error -0.2%)
    Weighted average = 137.5  (error +0.4%)


<a id="s11-ckm"></a>
## [§ 11 — CKM Matrix & CP Violation](#s11-ckm)

**Paper Appendix B.5.3 + technical T.10 §Berry Connection on Spin(RP³).**

The CP-violating phase is the Berry phase acquired by the spinor bundle $\mathrm{Spin}(\mathbb{RP}^3)$ around a Hopf-fiber loop on the parameter space $(k, |z|, \varphi)$:

$$\delta_{CP} = \pi\!\left(1 - \frac{1}{\varphi}\right) = \frac{\pi}{\varphi^2} \approx 1.200\,\text{rad} = 68.5°$$

Observed: $1.20 \pm 0.08$ rad ($68.4 \pm 3°$) — within $1\sigma$.

The Berry connection on $\mathrm{Spin}(\mathbb{RP}^3)$ decomposes as $A_{\rm radial}=0$, $A_{\rm scale}=0$, $A_{\rm Hopf}=1/2$ — the cascade coordinate $k$ is geometrically flat with respect to the spinor bundle, and all CP-violating content concentrates in the Hopf-fiber direction. The golden-ratio factor $\delta = \pi/\varphi^2$ enters through the self-similar $\tau$-action on the flag manifold of generations.

Cabibbo angle from mass ratios: $\theta_C \approx \sqrt{m_d/m_s}$ — overlap suppression from K\"ahler-radial separation, separate from the Berry-phase mechanism. CKM magnitudes $|V_{ij}|$ remain at the wavefunction-overlap (mass-ratio) approximation pending first-principles light-quark $|z|$ positions.

In [14]:
dcp = BP.delta_cp()
print("CP Violation Phase")
print("=" * 60)
print(f"  δ_CP = π(1 − 1/φ) = π/φ²")
print(f"       = {dcp['delta_cp_rad']:.5f} rad = {dcp['delta_cp_deg']:.2f}°")
print(f"  Observed: {dcp['observed_rad']:.2f} ± {dcp['observed_err_rad']:.2f} rad ({dcp['observed_deg']:.1f}°)")
print(f"  Within 1σ: {dcp['within_1sigma']}")
print(f"  Error: {dcp['error_pct']:+.2f}%")
print()

ckm = BP.ckm_angles()
print("CKM Angles")
print("-" * 40)
print(f"  θ_C (Cabibbo) = √(m_d/m_s) = {ckm['theta_cabibbo_deg']:.2f}°")
print(f"  |V_us| pred   = sin(θ_C) = {ckm['V_us_predicted']:.4f}")
print(f"  |V_us| obs    = {ckm['V_us_observed']:.4f}  (error {ckm['error_pct']:+.1f}%)")
print()

J = BP.jarlskog_invariant()
print("Jarlskog Invariant")
print("-" * 40)
print(f"  J (PPM δ_CP) = {J['J']:.3e}")
print(f"  J (observed) = {J['J_observed']:.3e}")
print(f"  Error:         {J['error_pct']:+.1f}%")


CP Violation Phase
  δ_CP = π(1 − 1/φ) = π/φ²
       = 1.19998 rad = 68.75°
  Observed: 1.20 ± 0.08 rad (68.8°)
  Within 1σ: True
  Error: -0.00%

CKM Angles
----------------------------------------
  θ_C (Cabibbo) = √(m_d/m_s) = 12.81°
  |V_us| pred   = sin(θ_C) = 0.2217
  |V_us| obs    = 0.2243  (error -1.1%)

Jarlskog Invariant
----------------------------------------
  J (PPM δ_CP) = 3.188e-05
  J (observed) = 3.080e-05
  Error:         +3.5%


<a id="s12-pmns"></a>
## [§ 12 — PMNS Matrix & Neutrino Sector](#s12-pmns)

**Paper §7 (SM Parameters).**

Tribimaximal (TBM) mixing from $Z_2 \times A_4$:
$\sin^2\theta_{12} = 1/3$, $\sin^2\theta_{23} = 1/2$, $\sin^2\theta_{13} = 0$.

$\theta_{\rm strong} = 0$ exactly — the $\theta$-term is $T$-odd, and the framework imposes $T$-invariance via the $\tau$-involution.

Sterile neutrino mass window: $k = 61$–$62$ brackets 3.5 keV X-ray line.

In [ ]:
pmns = NU.pmns_tribimaximal()
print("PMNS (Tribimaximal)")
print("=" * 60)
print(f"  sin²θ₁₂: PPM = {pmns['sin2_theta12_ppm']:.4f}  obs = {pmns['sin2_theta12_obs'][0]} ± {pmns['sin2_theta12_obs'][1]}  err = {pmns['theta12_error_pct']:+.1f}%")
print(f"  sin²θ₂₃: PPM = {pmns['sin2_theta23_ppm']:.4f}  obs = {pmns['sin2_theta23_obs'][0]} ± {pmns['sin2_theta23_obs'][1]}  err = {pmns['theta23_error_pct']:+.1f}%")
print(f"  sin²θ₁₃: PPM = {pmns['sin2_theta13_ppm']:.4f}  obs = {pmns['sin2_theta13_obs'][0]} ± {pmns['sin2_theta13_obs'][1]}  {pmns['theta13_status']}")
print(f"  Note: {pmns['note']}")
print()

ts = NU.theta_strong()
print(f"Strong CP: θ = {ts['theta']}  (observed bound: < {ts['observed_bound']})")
print(f"  Mechanism: {ts['mechanism']}")
print()

sn = NU.sterile_neutrino_mass_window()
print(f"Sterile ν mass window:")
print(f"  k={sn['k_upper']}: E = {sn['E_upper_keV']:.1f} keV (upper)")
print(f"  k={sn['k_lower']}: E = {sn['E_lower_keV']:.1f} keV (lower)")
print(f"  Brackets 3.5 keV X-ray line: {sn['brackets_xray']}")

nb = NU.neutrino_mass_bounds()
print(f"\nActive neutrino mass scale:")
print(f"  E(k=63) = {nb['E63_eV']:.3f} eV,  E(k=64) = {nb['E64_eV']:.4f} eV")


<a id="s13-heat-kernel"></a>
## [§ 13 — Heat Kernel Coefficients & Spectral Zeta](#s13-heat-kernel)

**Paper Appendix A, B.**

Seeley–DeWitt coefficients for scalar $\Delta$ on $\mathbb{CP}^3$:
$$a_0 = 1/48, \quad a_2 = 1/12, \quad a_4 = 1/6, \quad a_6 = 212/945$$

$$\zeta_\Delta(0) = -733/945 \approx -0.7757$$

$$\log\det(\Delta) = -\zeta'_\Delta(0) \approx 0.250$$

In [16]:
hk = S.heat_kernel_coefficients()
print("Heat Kernel Coefficients (scalar Δ on CP³)")
print("=" * 50)
for name, val in hk.items():
    if name != 'status':
        exact = {
            'a0': '1/48', 'a2': '1/12', 'a4': '1/6', 'a6': '212/945'
        }.get(name, '')
        print(f"  {name} = {val:.8f}  = {exact}")

print()
z0 = S.zeta_delta_0()
print(f"ζ_Δ(0) = {z0:.8f}  = -733/945 = {-733/945:.8f}")

print()
ld = S.log_det_delta()
dd = S.det_delta()
print(f"log det(Δ) = -ζ'_Δ(0) = {ld:.4f}")
print(f"det(Δ) = {dd:.4f}")

print()
z1 = S.Z1_oneloop()
print(f"Z₁ (one-loop) = {z1:.3f}  — all O(1), φ absent from perturbative spectrum")

print()
C_st = S.short_time_coefficient()
print(f"Short-time coefficient C = {C_st:.6f}  = 3√π/4 = {3*math.sqrt(math.pi)/4:.6f}")

print()
sep = S.separate_manifold_ratio()
print(f"Separate-manifold ratio: 1/α = {sep['alpha_inv_sep']:.3f}  (error {sep['error_pct']:+.3f}% at t={sep['t']:.5f})")


Heat Kernel Coefficients (scalar Δ on CP³)
  a0 = 0.02083333  = 1/48
  a2 = 0.08333333  = 1/12
  a4 = 0.16666667  = 1/6
  a6 = 0.22433862  = 212/945

ζ_Δ(0) = -0.77566138  = -733/945 = -0.77566138

log det(Δ) = -ζ'_Δ(0) = 0.2500
det(Δ) = 1.2840

Z₁ (one-loop) = 0.880  — all O(1), φ absent from perturbative spectrum

Short-time coefficient C = 1.329340  = 3√π/4 = 1.329340

Separate-manifold ratio: 1/α = 136.745  (error +0.212% at t=0.03183)


In [17]:
pt = S.poschl_teller_eigenvalues(10)
print("Pöschl-Teller Tube Eigenvalues")
print("=" * 40)
print(f"{'n':>3}  {'E_n':>12}")
print("-" * 18)
for n, E in enumerate(pt):
    print(f"{n:3d}  {E:12.3f}")


Pöschl-Teller Tube Eigenvalues
  n           E_n
------------------
  0        28.485
  1        79.799
  2       163.113
  3       278.426
  4       425.740
  5       605.054
  6       816.368
  7      1059.681
  8      1334.995
  9      1642.309


<a id="s14-instanton"></a>
## [§ 14 — Instanton Sector](#s14-instanton)

**Paper §8, Appendix A.**

Action: $S = (N{-}1) r^2 \pi = 30\pi \approx 94.248$

Zero modes: $n = 2(N^2{-}1) = 30 = \dim_\mathbb{R}(\text{PGL}(4,\mathbb{C}))$

Key match: $e^{-30\pi} \approx \varphi^{-196}$ (0.073% exponent mismatch)

$T^2$ partition function from Dedekind $\eta$: $\tau = i \cdot 10/\pi^2$

In [18]:
S_inst = I.instanton_action()
print("Instanton Action")
print("=" * 60)
print(f"  d = N-1 = {C.N_OUTCOMES - 1}")
print(f"  r² = 2(N+1) = {C.R_SQUARED}")
print(f"  S = d × r² × π = {C.N_OUTCOMES-1} × {C.R_SQUARED} × π = {S_inst:.4f}")
print(f"  30π = {30*math.pi:.4f}  ✓")
print()

zm = I.zero_mode_count()
print("Zero Modes")
print("-" * 40)
print(f"  h⁰(f*T_CP³) = {zm['n_complex']} complex = {zm['n_real']} real")
print(f"  dim_R(PGL(4,C)) = 2(N²-1) = 2×15 = {zm['dim_R_PGL4C']}  ✓")
print()

ck = I.phi_196_check()
print("φ^{-196} Match")
print("-" * 40)
print(f"  30π       = {ck['S_30pi']:.4f}")
print(f"  196·ln(φ) = {ck['exponent_phi196']:.4f}")
print(f"  Mismatch  = {ck['mismatch_pct']:.4f}%")
print(f"  e^{{-30π}}  = {ck['exp_neg_S']:.4e}")
print(f"  φ^{{-196}}  = {ck['phi_neg_196']:.4e}")
print(f"  Ratio     = {ck['ratio']:.4f}")


Instanton Action
  d = N-1 = 3
  r² = 2(N+1) = 10.0
  S = d × r² × π = 3 × 10.0 × π = 94.2478
  30π = 94.2478  ✓

Zero Modes
----------------------------------------
  h⁰(f*T_CP³) = 15 complex = 30 real
  dim_R(PGL(4,C)) = 2(N²-1) = 2×15 = 30  ✓

φ^{-196} Match
----------------------------------------
  30π       = 94.2478
  196·ln(φ) = 94.3175
  Mismatch  = 0.0740%
  e^{-30π}  = 1.1714e-41
  φ^{-196}  = 1.0925e-41
  Ratio     = 1.0722


In [19]:
tau_mod = I.t2_modular_parameter()
print("T² Modular Parameter")
print("=" * 60)
print(f"  β   = 10/π = {tau_mod['beta']:.6f}")
print(f"  πR  = π    = {tau_mod['piR']:.6f}")
print(f"  τ   = iβ/(πR) = i × {tau_mod['tau_imag']:.6f}")
print(f"  q   = e^{{-2πτ}} = {tau_mod['q']:.6e}")
print()

eta = I.dedekind_eta(tau_mod['tau_imag'])
print(f"  |η(τ)| = {eta:.8f}")
print()

zt2 = I.zt2_per_scalar()
print(f"Z_T² per scalar:")
print(f"  log Z_T² = {zt2['log_ZT2']:.4f}")
print(f"  Z_T²     = {zt2['ZT2']:.4f}")
print()

zt6 = I.zt2_total(6)
print(f"Z_T² total (6 dof):")
print(f"  log Z_T² = {zt6['log_ZT2_total']:.4f}")
print(f"  Z_T²     = {zt6['ZT2_total']:.4f}")
print()

pf = I.prefactor_subtotal()
print("Instanton Prefactor Budget")
print("-" * 40)
print(f"  log F_zero  = {pf['log_F_zero']:.3f}  (V_⊥^15 = 10^15)")
print(f"  log F_trans = {pf['log_F_trans']:.3f}  (translation moduli)")
print(f"  log Z_T²    = {pf['log_ZT2_total']:.3f}  (T² Dedekind η)")
print(f"  Subtotal    = {pf['subtotal']:.3f}")
print(f"  Target      ≈ {pf['target_log_J']:.0f}")
print(f"  Z_ws needed ≈ {pf['log_Z_worldsheet_needed']:.1f}  [PARKED/FFS]")


T² Modular Parameter
  β   = 10/π = 3.183099
  πR  = π    = 3.141593
  τ   = iβ/(πR) = i × 1.013212
  q   = e^{-2πτ} = 1.718682e-03

  |η(τ)| = 0.76568735

Z_T² per scalar:
  log Z_T² = 0.5274
  Z_T²     = 1.6945

Z_T² total (6 dof):
  log Z_T² = 3.1644
  Z_T²     = 23.6745

Instanton Prefactor Budget
----------------------------------------
  log F_zero  = 34.539  (V_⊥^15 = 10^15)
  log F_trans = 5.011  (translation moduli)
  log Z_T²    = 3.164  (T² Dedekind η)
  Subtotal    = 42.714
  Target      ≈ -100
  Z_ws needed ≈ -142.7  [PARKED/FFS]


<a id="s15-golden-ratio"></a>
## [§ 15 — Golden Ratio Investigation](#s15-golden-ratio)

**Paper section-gravity.tex (Why $\varphi$).**

Structural chain: $A_5 \to \mathbb{Q}(\sqrt{5}) \to \varphi$

$\mathfrak{sl}(4,\mathbb{R})|_{A_5} = \chi_1 \oplus 3\chi_3 \oplus \chi_5$ (dim check: $1 + 9 + 5 = 15$)

Dirichlet L-function: $L(1, (\cdot/5)) = 2\ln\varphi / \sqrt{5}$

In [20]:
a5 = GR_phi.a5_decomposition()
print("A₅ Decomposition of Instanton Moduli")
print("=" * 60)
print(f"  sl(4,R)|_A₅ = {a5['decomposition']}")
print(f"  dim check: {a5['total_dim']} = {a5['dim_sl4R']}  ✓")
print(f"  Character field: {a5['character_field']}")
print(f"  Fundamental unit: φ = {a5['fundamental_unit']:.6f}")
print()

L = GR_phi.dirichlet_l_function()
print(f"Dirichlet L-function:")
print(f"  L(1,(·/5)) = 2ln(φ)/√5 = {L['L_1_chi5']:.8f}")
print()

print("Chain: A₅ → Q(√5) → φ")
print("  A₅ acts on moduli (icosahedral subgroup of PGL(4,C))")
print("  Character field of A₅ = Q(√5)")
print(f"  φ = (1+√5)/2 = {C.PHI:.10f}")


A₅ Decomposition of Instanton Moduli
  sl(4,R)|_A₅ = χ₁ ⊕ 3·χ₃ ⊕ χ₅
  dim check: 15 = 15  ✓
  Character field: Q(√5)
  Fundamental unit: φ = 1.618034

Dirichlet L-function:
  L(1,(·/5)) = 2ln(φ)/√5 = 0.43040894

Chain: A₅ → Q(√5) → φ
  A₅ acts on moduli (icosahedral subgroup of PGL(4,C))
  Character field of A₅ = Q(√5)
  φ = (1+√5)/2 = 1.6180339887


<a id="s16-cosmology"></a>
## [§ 16 — Cosmological Predictions](#s16-cosmology)

**Paper §10–11 (Gravitational Constants, Cosmological Evidence).**

$\Lambda$ is fixed by the *static* boundary capacity $N_\infty = \varphi^{392}$:

$$\Lambda = \frac{2(m_\pi c^2)^2}{(\hbar c)^2 N_\infty}$$

The Hubble parameter is the inverse of cosmic age, computed from the Sidharth relations: $R_{\rm universe} = \sqrt{N_\infty}\,\lambda_C$, $T = R/c$, $H_0 = 1/T$.

$G_{\rm eff}(z) = G_0 (1+z)^{3/2}$ in the holographic-screening regime: the cumulative actualization count $M(t)$ — distinct from the static $N_\infty$ — sets the screening level, and $M(z) \propto a^3 \propto (1+z)^{-3/2}$ in matter domination gives the exponent. Falsifiable against early-galaxy observations (JWST z~10–12).

Dark energy equation of state: backreaction from actualization gives $w_0 = -1 + \beta\epsilon/(1+\beta\epsilon)$, $w_a = 3\beta\epsilon/(1+\beta\epsilon)^2$ with $\epsilon = \Omega_m/\Omega_\Lambda$; $\beta$ is OPEN (FFS calculation).

In [21]:
lam_cc = GR.cosmological_constant()
print("Cosmological Constant")
print("=" * 60)
print(f"  N = φ^392 = {lam_cc['N']:.4e}")
print(f"  Λ (PPM)  = {lam_cc['Lambda_m2']:.4e} m⁻²")
print(f"  Λ (obs)  = {lam_cc['Lambda_obs']:.1e} m⁻²")
print(f"  Error    = {lam_cc['error_pct']:+.1f}%")
print()

h0 = GR.hubble_from_age()
print("Hubble Constant")
print("-" * 40)
print(f"  T = {h0['T_universe_Gyr']:.3f} Gyr")
print(f"  H₀ = 1/T = {h0['H0_km_s_Mpc']:.1f} km/s/Mpc")
print(f"  Planck: 67.4 ± 0.5, SH0ES: 73.0 ± 1.0")
print()

hs = GR.hubble_from_sidharth()
print("Sidharth Relations (from N = φ^392)")
print("-" * 40)
print(f"  √N     = φ¹⁹⁶ = {hs['sqrt_N']:.4e}")
print(f"  λ_C    = {hs['lambda_C_m']:.4e} m")
print(f"  R      = √N·λ_C = {hs['R_universe_m']:.4e} m")
print(f"  T      = {hs['T_universe_Gyr']:.2f} Gyr")
print(f"  H₀     = {hs['H0_km_s_Mpc']:.1f} km/s/Mpc")


Cosmological Constant
  N = φ^392 = 8.3783e+81
  Λ (PPM)  = 1.1169e-52 m⁻²
  Λ (obs)  = 1.1e-52 m⁻²
  Error    = +1.5%

Hubble Constant
----------------------------------------
  T = 13.797 Gyr
  H₀ = 1/T = 70.9 km/s/Mpc
  Planck: 67.4 ± 0.5, SH0ES: 73.0 ± 1.0

Sidharth Relations (from N = φ^392)
----------------------------------------
  √N     = φ¹⁹⁶ = 9.1533e+40
  λ_C    = 1.4620e-15 m
  R      = √N·λ_C = 1.3382e+26 m
  T      = 14.14 Gyr
  H₀     = 69.1 km/s/Mpc


In [22]:
print("G_eff(z) = G₀(1+z)^{3/2}")
print("=" * 40)
print(f"{'z':>5}  {'G_eff/G₀':>10}  {'δ_c^PPM':>10}")
print("-" * 30)
for z in [0, 1, 2, 5, 8, 10, 12]:
    print(f"{z:5.0f}  {GR.g_eff(z):10.2f}  {GR.delta_c_ppm(z):10.3f}")

print()
print("GW Dispersion")
print("-" * 40)
for f, label in [(100, "LIGO"), (1e10, "PTA"), (1e15, "UHE")]:
    gw = GR.gw_dispersion(f)
    print(f"  {label:>6} ({f:.0e} Hz): Δv/c = {gw['delta_v_over_c']:.2e}")

# Plot G_eff(z)
z_arr = np.linspace(0, 15, 200)
g_arr = np.array([(1+z)**1.5 for z in z_arr])

fig, ax = plt.subplots(figsize=(10, 5))
ax.plot(z_arr, g_arr, 'b-', linewidth=2.5, label=r'$G_{\rm eff}(z) = G_0(1+z)^{3/2}$')
ax.axhline(1, color='gray', linestyle='--', alpha=0.5)

# Mark key points
for z_mark, label in [(6, 'z=6: reionization'), (10, 'z=10: early galaxies'), (12, 'z=12: JWST')]:
    ax.plot(z_mark, (1+z_mark)**1.5, 'ro', markersize=8)
    ax.annotate(f'{label}\n$G_{{eff}}={((1+z_mark)**1.5):.0f}G_0$',
                xy=(z_mark, (1+z_mark)**1.5), xytext=(z_mark+1, (1+z_mark)**1.5+5),
                fontsize=9, arrowprops=dict(arrowstyle='->', color='red'))

ax.set_xlabel('Redshift z', fontsize=12)
ax.set_ylabel(r'$G_{\rm eff}(z) / G_0$', fontsize=12)
ax.set_title(r'PPM: Effective Gravitational Constant Evolution — Falsifiable Prediction', fontsize=12)
ax.legend(fontsize=11)
ax.grid(True, alpha=0.3)
plt.tight_layout()
plt.show()


G_eff(z) = G₀(1+z)^{3/2}
    z    G_eff/G₀     δ_c^PPM
------------------------------
    0        1.00       1.753
    1        2.83       1.537
    2        5.20       1.423
    5       14.70       1.248
    8       27.00       1.155
   10       36.48       1.112
   12       46.87       1.077

GW Dispersion
----------------------------------------
    LIGO (1e+02 Hz): Δv/c = 5.71e-82
     PTA (1e+10 Hz): Δv/c = 5.71e-66
     UHE (1e+15 Hz): Δv/c = 5.71e-56


In [23]:
print("Dark Energy Equation of State")
print("=" * 50)
for ratio in [0.01, 0.03, 0.05, 0.10]:
    w = GR.w_eff(ratio)
    print(f"  Ω_δ/Ω_DE = {ratio:.2f}  →  w_eff = {w:.4f}")

print()
print("Note: β = Ω_δ/Ω_DE in the backreaction formula is OPEN (requires FFS).")
print("If DESI w₀≈-0.55, w_a≈-1.75 holds, the simple β-model is falsified.")


Dark Energy Equation of State
  Ω_δ/Ω_DE = 0.01  →  w_eff = -0.9933
  Ω_δ/Ω_DE = 0.03  →  w_eff = -0.9800
  Ω_δ/Ω_DE = 0.05  →  w_eff = -0.9667
  Ω_δ/Ω_DE = 0.10  →  w_eff = -0.9333

Note: β = Ω_δ/Ω_DE in the backreaction formula is OPEN (requires FFS).
If DESI w₀≈-0.55, w_a≈-1.75 holds, the simple β-model is falsified.


<a id="s16b-consciousness"></a>
## [§ 16b — Consciousness Numerics](#s16b-consciousness)

**Paper §12 (Consciousness Locator). Technical: ch13.**

The $\tau$-firing scale at biological temperature places the consciousness window at $k \approx 75$ (where $E(k) \approx k_B T_{\rm bio}$, $T = 310$ K). Integration time and motor-reliability redundancy are downstream consequences of this single scale assignment.

**Anti-woo:** these are conditions biology *exploits*, not conditions the framework *produces*. Active inference is a property of organized matter satisfying the $R \approx 1$, $\Phi > 0$, two-loop conditions; the variational principle does not produce consciousness.

In [24]:
k_c = GR.k_conscious(310)
print(f"  k_conscious(310 K) = {k_c:.2f}")
print(f"  E(k_c) = {H.energy_mev(k_c)*1e3:.1f} µeV  ≈ k_BT(310K) = 26.7 meV")

ti = GR.integration_time(310)
print(f"\n  Per-event timescale (single τ-event):")
print(f"    τ_sys (collective, 10^14 sites) = {ti['tau_sys_eff_s']:.3e} s")
print(f"    τ_bath (thermal)               = {ti['tau_bath_thermal_s']:.3e} s")
print(f"\n  Integration time (collective, F9 form):")
print(f"    t_integrate = τ_sys² / τ_bath = {ti['t_integrate_ms']:.3f} ms")
print(f"    Single-site form (legacy):     {ti['t_integrate_singlesite_ms']:.4f} ms")

nr = GR.n_reliable()
print(f"\n  N_reliable = {nr['N_reliable']:.2e}  (corticospinal-tract scale 10⁵–10⁶)")


  k_conscious(310 K) = 75.35
  E(k_c) = 0.0 µeV  ≈ k_BT(310K) = 26.7 meV

  Per-event timescale (single τ-event):
    τ_sys (collective, 10^14 sites) = 2.464e-07 s
    τ_bath (thermal)               = 1.000e-12 s

  Integration time (collective, F9 form):
    t_integrate = τ_sys² / τ_bath = 60.710 ms
    Single-site form (legacy):     0.1245 ms

  N_reliable = 4.85e+05  (corticospinal-tract scale 10⁵–10⁶)


<a id="falsification-tests"></a>
## [Falsification Tests](#falsification-tests)

Six fixed-comparison tests (F1–F6) that change one structural commitment at a time and compute what the framework would then predict have moved to the public-facing **[predictions.ipynb](predictions.ipynb)**, in the *How to Break the Framework* section. They sit there alongside the live sliders so the structural argument is in one place.